src/components/APICatalog/APICatalog.tsx

```diff
@@ -1,52 +1,51 @@
 import React from "react";
 import { useHistory } from "react-router";
-import { Avatar, Box, Grid, Paper, Typography, useTheme, useTranslation } from "@apisuite/fe-base";
+import { Avatar, Box, Grid, Typography, useTheme, useTranslation } from "@apisuite/fe-base";
 
 import { Tag } from "components/Tag";
+import { ApplicationCard } from "components/ApplicationCard/ApplicationCard";
 import useStyles from "./styles";
 import { APICatalogProps, APIDetails } from "./types";
 
 import { testIds } from "testIds";
 
-const APICatalog: React.FC<APICatalogProps> = ({ apisToDisplay }) => {
+const APICatalog: React.FC<APICatalogProps> = ({ apisToDisplay, limit }) => {
   const classes = useStyles();
   const { palette } = useTheme();
   const history = useHistory();
   const { t } = useTranslation();
 
   const handleOnCardClick = (details: APIDetails) => () => {
-    history.push(`/api-products/details/${details.id}/spec/${details.apiRoutingId}`);
+    history.push(`/api-products/details/${details.id}/spec/${details.apiRoutingId || 0}`);
```

Comment:
it's preferable to check if `apiRoutingId` is a valid number and only push to history if it is.

Reply:
A new ticket will need to be created for that. To also disable the over/click functionality if no click action is set.

Reply:
ok